fineract-provider/src/main/java/org/apache/fineract/portfolio/note/service/NoteWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -456,4 +472,14 @@ private Note getNoteForDelete(final JsonCommand command) {
         return noteForUpdate;
     }
 
+    public Map<String, Object> update(final JsonCommand command, Note noteForUpdate) {
+        final Map<String, Object> actualChanges = new LinkedHashMap<>(7);
+        final String newNote = command.stringValueOfParameterNamed("note");
+        if (!noteForUpdate.getNote().equals(newNote)) {
```

Comment:
are you sure the equals() will catch any content changes inside a Note? We might need a unit test for this logic as well.

Reply:
This is the same old logic, just moved from the entity to the service layer for better separation of concerns. 

The `equals()` method will work as will correctly identify changes in the note content, as it compares the entire string.